### PR TITLE
Symfony 4 support

### PR DESCRIPTION
**Pull request by istvancsabakis**
#SymfonyConHackday2017

### CODE REVIEW

**Comment by bytehead** [comment]
Thanks for contributing! I'll place some comments where needed :)

File: Tests/App/AppKernel.php [hunk 1]
```diff
@@ -1,5 +1,7 @@
 <?php
 
+namespace Oneup\UploaderBundle\Tests;
```
Comment (bytehead) [inline Tests/App/AppKernel.php]: This should be `namespace Oneup\UploaderBundle\Tests\App;`

File: phpunit.xml.dist [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 <phpunit bootstrap="./Tests/bootstrap.php" colors="true">
     <php>
-        <server name="KERNEL_DIR" value="Tests/App" />
```
Comment (bytehead) [inline phpunit.xml.dist]: `KERNEL_DIR` must not be removed, still used for Symfony 2.x and 3.x tests.

File: composer.json [hunk 3]
```diff
@@ -16,35 +16,36 @@
...
+        "symfony/browser-kit": "2.*|^3.0|^4.0",
+        "symfony/class-loader": "2.*|^3.0|^4.0",
+        "symfony/security-bundle": "2.*|^3.0|^4.0",
         "twistor/flysystem-stream-wrapper": "^1.0"
     },
 
     "suggest": {
         "knplabs/knp-gaufrette-bundle": "0.1.*",
-        "oneup/flysystem-bundle": "^1.2",
+        "oneup/flysystem-bundle": "^3.0",
```
Comment (bytehead) [inline composer.json]: Be aware: to keep it compatible with Symfony 2, 3, and 4 as well as PHP 5.6 and 7.x the constraint should be `^1.2|^2.0|^3.0` :)

Comment (bytehead) [reply composer.json]: Sorry, didnt' see it's only a suggest. Then you're right!

File: composer.json [hunk 4]
```diff
@@ -27,7 +27,7 @@
     "require-dev": {
         "amazonwebservices/aws-sdk-for-php": "1.5.*",
         "knplabs/gaufrette": "0.2.*@dev",
-        "oneup/flysystem-bundle": "^3.0",
+        "oneup/flysystem-bundle": "^1.2|^3.0",
```
Comment (bytehead) [inline composer.json]: Could also be compatible with version `2.x` of the `oneup/flysystem-bundle`:
` "oneup/flysystem-bundle": "^1.2|^2.0|^3.0",`

File: composer.json [hunk 5]
```diff
@@ -16,7 +16,7 @@
 
     "require": {
         "php":">=5.4",
-        "symfony/framework-bundle": "^2.4|^3.0|^4.0",
+        "symfony/framework-bundle": "4.0.*",
```
Comment (bytehead) [inline composer.json]: We should still be compatible with Symfony 2.x and 3.x
`^2.4|^3.0|^4.0` will be the right constraint.

**Comment by bytehead** [comment]
The tests fail probably due to https://github.com/symfony/symfony/issues/24524.

**Comment by istvancsabakis** [comment]
In symfony 4 all services will be private. We need to make OneupUploaderBundle services public by default or refractor services to remove container->get calls?

**Comment by bytehead** [comment]
I'll close this in favour of #300. Thanks for the PR!

**Pull request closed without merging**